Add edit and deactivate member features

gisell-royg-itti opened the pull request.

gisell-royg-itti committed:
Agrega modales para editar y desactivar un socio

add_edit_member_feature Agrega modales para editar y desactivar un socio

gisell-royg-itti committed:
Modifica llamada al servicio de editar y desact

add_edit_member_feature Modifica llamada a los servicios de editar y
desactivar.

gisell-royg-itti committed:
Merge branch 'master' into add_edit_member_feature

gisell-royg-itti committed:
Merge branch 'master' into add_edit_member_feature

# Conflicts:
#	components/members/MemberList.js

rodrigojv commented on components/members/DeactivateModal.js:
```diff
@@ -0,0 +1,79 @@
+  Button,
+  Text, useToast
+} from "@chakra-ui/react";
+import useDeactivateMember from "../../utils/useDeactivateMember";
```
Este import se puede hacer directo desde el directorio raíz `from "utils/useDeactivateMember"`

gisell-royg-itti replied:
Modificado.

rodrigojv commented on components/members/EditModal.js:
```diff
@@ -0,0 +1,342 @@
+  };
+
+  const setNewFormValues = (form) => {
+    values.name = form.name;
+    values.surname = form.surname;
+    values.email = form.email;
+    values.document = form.document;
+    values.birthdate = form.birthdate
+    values.sexo = form.sexo;
+    values.departamento = form.departamento;
+    values.city = form.city;
+    values.cellphone = form.cellphone;
+    values.razonsocial = form.razonsocial;
+    values.nfantasia = form.nfantasia;
+    values.ruc = form.ruc;
+    values.rubro = form.rubro;
+    values.empleados = form.empleados;
+    values.sitioweb = form.sitioweb;
+    values.facturacion = form.facturacion;
```
Para esto caso hay que ver que se puede hacer, capaz retornar `setValues` que está definido en `useForm` y luego en esta función hacer algo como: `setValues({...form})`

gisell-royg-itti replied:
Modificado. Pude utilizar la funcion ```updateValue```del useForm dentro de cada item del formulario.

rodrigojv commented on components/members/EditModal.js:
```diff
@@ -0,0 +1,342 @@
+  });
+
+  const initialValues = {
+    name: member.name != undefined ? member.name : "",
```
Acá se podría usar el optional chaining de JS `member?.name || ""` lo mismo para el resto de los campos en la incialización.

gisell-royg-itti replied:
Modificado.

gisell-royg-itti committed:
Modifica forma de actualizar valores al editar

add_edit_member_feature Modifica forma de actualizar valores en el
formulario de edición de datos del socio.
Se utiliza la funcion 'updatevalue' del useForm.
